### PR TITLE
Dynamic language docs

### PR DESCRIPTION
**Pull request by SerKnight**
Ruby Codegen Work

* Adding enums in the model
* Adds a couple comments
* Adds x-ruby-payload and x-ruby-requestBody vendor extension(strings)
  * for all POST's that require body parameters and fixes docs to use the new enum values - currently only accounting docs are fixed.

> Note, was not able to fix the specs in the mustache templates without an un-reasonable amount of scripting.. So will just add those & fix manually as new sets are layered in.

--- Needs Review ----- @SidneyAllen 
* pattern for adding dynamic language docs.. (Ruby is not able to be generated by htmlDocs2 at this point 👎  I joined their slack and am seeing if it is something I could open source contribute, or if there is another technical language reason why not..)
* Removed OrganisationEntityType enums as they were duplicates, and was causing runtime warnings in model due to duplication.

🚨 🚨 🚨 Need confirmation not breaking change in other SDK gen changing `OrganisationEntityType` to a string 🚨 🚨 🚨 

If breaking change I can roll back. Dup constants in `organisation.rb` not breakage but just requires more bash scripting.

> REF: https://github.com/XeroAPI/xero-ruby/pull/49

### CODE REVIEW

File: accounting-yaml/xero_accounting.yaml [hunk 1]
```diff
@@ -6037,7 +5999,6 @@ contact: {}, date:"2020-01-01", user:{} } ] } ] }'
         - in: query
           name: Statuses
           description: Filter by a comma-separated list Statuses. For faster response times we recommend using these explicit parameters instead of passing OR conditions into the Where filter.
-          style: simple
```
Comment (SidneyAllen) [inline accounting-yaml/xero_accounting.yaml:6040]: @SerKnight are we sure removing this won't cause problems for other SDKs?

File: accounting-yaml/xero_accounting.yaml [hunk 2]
```diff
@@ -11931,7 +11847,7 @@ contact: {}, date:"2020-01-01", user:{} } ] } ] }'
           $ref: '#/components/responses/400Error'
       requestBody:
         $ref: '#/components/requestBodies/historyRecords'
-  '/Quotes/{QuotesID}/pdf':
+  '/Quotes/{QuoteID}/pdf':
```
Comment (SidneyAllen) [inline accounting-yaml/xero_accounting.yaml:11850]: @SerKnight - was this a bug reported in getting Quote as PDF?

File: accounting-yaml/xero_accounting.yaml [hunk 3]
```diff
@@ -20690,19 +20594,6 @@ components:
         OrganisationEntityType:
           description: Organisation Type
           type: string
-          enum:
```
Comment (SidneyAllen) [inline accounting-yaml/xero_accounting.yaml:20693]: @SerKnight - we've already discussed.

Comment (SerKnight) [reply accounting-yaml/xero_accounting.yaml:20693]: Put enums for entity type back in

**Comment by SerKnight** [comment]
@SidneyAllen - I'm okay with not merging this master until I have a better grasp on how much work it would be rolling the ruby examples into the htmldocs2? Lmk your thoughts and I can totally keep generating ruby off this branch.

**Comment by SidneyAllen** [comment]
@SerKnight - I am comfortable letting you continue to iterate on this approach before deciding to merge or not merge.

**Comment by SerKnight** [comment]
Decided to merge this into master. It will likely be removed in the future in lui of a more consistent documentation approach, but that is TBD on an exact timeframe. Will not effect continued dev of all other sdks.

**Pull request merged**